library/xml/src/modules/XML.rb

```diff
@@ -293,12 +282,13 @@ def parse_node(node, result)
     # @param [Nokogiri::XML::Document] doc
     # @param [Hash] metadata for current doc
     # @param [Nokogiri::XML::Node] parent
-    # @param [Hash] content to write
+    # @param [Hash] contents content to write
+    # @param [Array] parent_array (for better error reporting)
     # @return [void]
-    def add_element(doc, metadata, parent, contents)
+    def add_element(doc, metadata, parent, contents, parent_array = nil)
```

Comment:
np, I would use named parameter instead of optional. This will help if we need more parameters in future.